share/lib/python/neuron/rxd/generalizedReaction.py

```diff
@@ -161,14 +161,16 @@ def spfilter(sp):
         ]
 
         sp_regions = None
-        if self._trans_membrane and (
-            sources or dests
-        ):  # assume sources share common regions and destinations share common regions
-            sp_regions = list(
-                {sptr()._region() for sptr in sources}.union(
-                    {sptr()._region() for sptr in dests}
+        if self._trans_membrane:
+            if sources or dests:
+                # assume sources share common regions and destinations share common regions
+                sp_regions = list(
```

Comment:
I know this is how it was before, but this could be consolidated as:

```python
sp_regions = list({sptr()._region() for sptr in sources + dests})
```